app/data_model/answer_store.py

```diff
@@ -0,0 +1,192 @@
+    def __init__(self, existing_answers=None):
+        self.answers = existing_answers or []
+
+    @staticmethod
+    def _validate(answer):
+        if type(answer) is not Answer:
+            raise TypeError("Method only supports Answer argument type")
+
+    def add(self, answer):
+        """
+        Add a new answer into the answer store.
+
+        :param answer: A dict of flattened answer details.
+        """
+        self._validate(answer)
+
+        answer_to_add = answer.__dict__.copy()
+
+        if not self.exists(answer):
+            self.answers.append(answer_to_add)
+
+    def update(self, answer):
+        """
+        Update the value of an answer already in the answer store.
+
+        :param answer: A dict of flattened answer details.
+        """
+        self._validate(answer)
+
+        for index, existing in enumerate(self.answers):
+            if answer.equals_dict(existing):
+                self.answers[index]['value'] = answer.value
+                break
+
+    def add_or_update(self, answer):
+        """
+        Add a new answer into the answer store, or update if it exists.
+
+        :param answer: An answer object.
+        """
+        self._validate(answer)
+
+        if self.exists(answer):
+            self.update(answer)
+        else:
+            self.add(answer)
+
+    def get(self, answer):
+        """
+        Returns the value of an answer.
+
+        :param answer: The ids for the answer ro return
+        :return: The value of the answer found
+        """
+        self._validate(answer)
+
+        for existing in self.answers:
+            if answer.equals_dict(existing):
+                return existing['value']
+
+        return None
+
+    def exists(self, answer):
+        """
+        Checks to see if an answer exists in the answer store.
+
+        :param answer: A dict of flattened answer details.
+        :return: True if the answer is in the store, False if not.
+        """
+        self._validate(answer)
+
+        return self.count(answer) > 0
+
+    def count(self, answer):
+        """
+        Count of the number of instances of an answer in the answer store.
+
+        :param answer: A dict of flattened answer details.
+        :return: 0 if the answer doesn't exist, otherwise the number of instances.
+        """
+        self._validate(answer)
+
+        return len(self.filter(answer.answer_id, answer.block_id, answer.answer_instance))
+
+    def filter(self, answer_id=None, block_id=None, answer_instance=None):
+        """
+        Find all answers in the answer store for a given set of filter parameter matches.
+
+        :param answer_id:
+        :param block_id:
+        :param answer_instance:
+        :return:
+        """
+        filtered = []
+        filter_vars = {
+            "answer_id": answer_id,
+            "block_id": block_id,
+            "answer_instance": answer_instance,
+        }
+        for answer in self.answers:
+            matches = True
```

Comment:
Do we need to define this bool? on the first match can't we just append the answer and continue?